Rewrite

s0lst1ce opened the pull request:
completely rewrote ForeBot by moving all of its functions to cogs which were then put in extensions. This PR will allow much more efficient collaboration because suites of functions are now separated in distinct files but also because the overall quality of the code has improved. Finally it allows the bot to be updated without ever having to get it offline. Only reload an extension using the appropriate function and the bot will be updated.

s0lst1ce committed:
put most functions in cogs

s0lst1ce committed:
cleaned code and removed token

s0lst1ce committed:
added check for self user in the Poll cog

s0lst1ce committed:
moved all cogs to extensions

s0lst1ce committed:
fixed #8 & removed token

NotaSmartDev commented:
@adrienluitot I think we should probably document the new code before actually merging the PR.

NotaSmartDev commented:
#13 I had forgotten to add `logging` support. I'll fix this asap.

s0lst1ce committed:
added safe extension loading

s0lst1ce committed:
added basic logging support

adrienluitot commented on Embedding.py:
```diff
@@ -0,0 +1,34 @@
+	@commands.command()
+	async def embed(self, ctx, *args):
+		"""allows you to post a message as an embed. Your msg will be reposted by the bot as an embed !"""
+		if ctx.channel.name in POLL_ALLOWED_CHANNELS:
```
Maybe we should use `channel.id` instead of `channel.name`

s0lst1ce replied:
See #4

adrienluitot commented on Poll.py:
```diff
@@ -0,0 +1,48 @@
+
+		#checking that user isn't the bot
+		print(reaction.message.channel.name)
+		if user != self.bot and (reaction.message.channel.name in POLL_ALLOWED_CHANNELS):
```
Same, maybe we should use `channel.id`

adrienluitot commented on Slapping.py:
```diff
@@ -0,0 +1,61 @@
+
+	@commands.command()
+	@commands.has_any_role(*GESTION_ROLES)
+	async def pardon(self, ctx, member:discord.Member):
```
Maybe add the possibility to remove only 1, 2, `x` warnings

s0lst1ce replied:
I let you open an issue for this then

adrienluitot commented on Poll.py:
```diff
@@ -0,0 +1,48 @@
+from checks import *
+
+
+class Poll(commands.Cog):
```
Add a command to add poll channel if has permission to

s0lst1ce replied:
See #4

adrienluitot commented on settings.py:
```diff
@@ -0,0 +1,41 @@
+
+PREFIX = "::"
+TOKEN = "your_token"
+AUTHOR_ID=289426079544901633
```
Shouldn't be `AUTHORS_ID` ? :roll_eyes:

s0lst1ce replied:
It's rather meant to be the ID of the one running the script. I'll change the name so that it's more appropriate.

adrienluitot replied:
Okay :D

adrienluitot commented on settings.py:
```diff
@@ -0,0 +1,41 @@
+	"shrug": "\U0001f937",
+}
+
+ADMIN_ROLE = ["Server Admin", "Bot Admin"]
```
`ADMIN_ROLES`*

adrienluitot commented on settings.py:
```diff
@@ -0,0 +1,41 @@
+	"thumbsdown": "\U0001f44e",
+	"shrug": "\U0001f937",
+}
+
```
Maybe add a command to add or remove roles in `ADMIN_ROLE(S)`, `GESTION_ROLES`, `DEV_ROLES`

s0lst1ce replied:
See #4

adrienluitot reviewed (approved):
Most of my comments are enhancement, but I think for channel.id instead of channel.name, we should do this quickly to avoid problem :smile:

adrienluitot commented on Poll.py:
```diff
@@ -0,0 +1,48 @@
+		self.bot = bot
+		
+	@commands.Cog.listener()
+	async def on_reaction_add(self, reaction, user):
```
switch to `on_raw_reaction_add` ?

s0lst1ce replied:
Completely forgot to do this.

s0lst1ce replied:
This will be added once #9 will be merged

adrienluitot commented on settings.py:
```diff
@@ -0,0 +1,32 @@
+	GESTION_ROLES.append(role)
+	DEV_ROLES.append(role)
+
+SLAPPED_LOG_FILE = "slapped.txt"
```
I think it would be better to put constants like that in concerned files (slapping.py for thi by exemple), it will reduce problems with merge and if we want "everyone" to be able to add cogs I think it's better

s0lst1ce replied:
I don't really see why. To me it's a nice way to be know immediately where the info is and have a nice way to change it. Particularly useful when #4 will be done. Now I stay open to your arguments and I may change my mind ;)

adrienluitot replied:
In fact it's because by this way if someone just want to add the code, or re use only one part (like slapping without todo) he'll have some code for nothing. But it's also easier and less restrictive if someone want to enhance the bot and add a new part.

s0lst1ce replied:
never saw it that way but it's true. We should talk about it tomorrow. Both ideas have potential we just need to determine with precision the objectives of this bot.

Commit:
add todo (need review)

Commit:
improved the todo bot and fixed some bugs

also added gitignore
(Corrections and enhancements are all based on @NotaSmartDev 's comment :D)

s0lst1ce committed:
Merge pull request #14 from organic-bots/adrien

A `todo` suite of commands were added thanks to @adrienluitot . This allows you to keep your planning tidy directly from discord. Especially useful if you use discord as your main platform for development.

s0lst1ce committed:
added logging support to all files

s0lst1ce committed:
Merge branch 'rewrite' of https://github.com/organic-bots/ForeBot into rewrite

s0lst1ce committed:
fixed minor git tracking issues

s0lst1ce committed:
Poll: switched from using channel/guild/member names to ids

NotaSmartDev commented:
Now using IDs instead of names for guild/channels/members

s0lst1ce committed:
added logging support for Todo

NotaSmartDev commented:
Fixed #13 by adding logging to all extensions.

NotaSmartDev commented:
I think we've added everything that was needed for #9 so unless @adrienluitot has any objection I'll merge.

s0lst1ce committed:
Poll: minor API's use improvement & Todo: fixed scopes issues

s0lst1ce committed:
added #12

s0lst1ce committed:
may have fixed #11

NotaSmartDev commented:
Should #4 be done before we merge #9 ?

s0lst1ce committed:
fixed #15

s0lst1ce committed:
added pardon count (#16)

s0lst1ce committed:
added a clear command (#17) & fixed some minor issues

s0lst1ce committed:
embed command now supports images (#6)

s0lst1ce committed:
users can now delete their polls (#5) & made poll functions more flexible

NotaSmartDev commented:
This was a huge update I'm happy to pronounce complete ! Its main purpose has been achieved. We've been able to completely rewrite the entire code to make it more pythonic. This major rewrite now allows one to update the code without taking the bot offline by using the smart extension principle of the API. It is also much easier to find out bugs happening to the bot for a `logging` has been fully implemented to the bot and is easily extendable. This was one of the main purpose: make any future addition much easier to both build and implement. This should also allow a much wider range of contributors to help us in the development. This branch doesn't solve everything though, there are still efforts required toward documentation (see #1 ) and repository structure. But this is already an important milestone which paves the way to all future updates. Finally a number of issues have been fixed in this update as you can see in the following change log:
- admins can now update extensions on the fly with a new & safe `ext` suite of functions (#8 )
- added smart and extensive `logging` support (#13 )
- added a new suite of commands that allow smart task list management using labels directly from discord
- changed global settings from using unsafe names to unique snowflake discord IDs
- `poll` colors are now dynamic and change with every new reaction (#12 )
- changed the design of embeds to give more credits to authors (#11 )
- fixed a bug where users could have multiple opposite reactions to the same poll (#15 )
- gave more freedom to admins by letting them choose how much they wish to `pardon` a user (#16 )
- added the `clear` command that lets admins delete a set amount of messages to ease moderation (#17 )
- embeds now support images (#6 )
- users can now delete their own polls which prevents admin from having to clean mistakes themselves (#5 )
